content/specific-skill-success-criteria/classes-and-objects/_index.md

```diff
@@ -23,3 +23,122 @@ All students need to provably understand all of the following concepts:
+  def __init__(self, acc_no, amount):
+    self.acc_no = acc_no
+    self.amount = amount
+```
+
+## Instantiation
+Instantiating a class is creating a copy of the class which inherits all class variables and methods. We simply call the class as if it were a function, passing the arguments that the `__init__()` method defines (except `self`).
+```
+saitama_acc = BankAccount("sa1234", 10000)
+```
+
+A class can have multiple instances.
+```
+kuroko_acc = BankAccount("ka1234", 300)
+tsukasa_acc = BankAccount("ta1234", 75010)
+vash_acc = BankAccount("va1234", 813650)
+```
+
+
+## Getters and setters
+The main purpose that getters and setters serve in object-orientated programs is for ensuring data encapsulation - or put another way, as a mechanism for hiding the implementation details of a class from the user. In Python, getters and setters are not the same as for other object-orientated languages. Private variables in Python are not actually hidden fields. In Python getters and setters are typically used when:
+
+- we want to add validation logic for getting and setting a value
+- to avoid the direct access of a class field so that private variables cannot be accessed directly/modified by an outside user
+
+### Getters
+A getter is a method that gets the value of a property.
+```
+  def balance(self):
+    return self.amount
+```
+
+```
+saitama_acc.amount
+```
+
+```
+saitama_acc.balance()
+```
+
+### Setters
+A setter is a method that sets the value of a property.
+```
+  def deposit(self, amount):
+    self.amount += amount
+
+  def withdraw(self, amount):
+    self.amount -= amount
+```
+
+```
+saitama_acc.deposit(500)
+saitama_acc.withdraw(320)
+```
+
+
+## Inheritance
+Inheritance enables us to create a new class from an existing class. The new class (child class) is a specialized version of the existing class (parent class) and it inherits all the non-private variables and methods of the existing class.
+
+If no additional instance variable(s) are required in the child class, `__init__()` can be left out.
+```
+class SavingsAccount(BankAccount):
+  pass
+```
+
+If additional instance variable(s) are required in the child class, `super()` is used.
+```
+class SavingsAccount(BankAccount):
+    def __init__(self, acc_no, amount, main_acc_instance):
+        super().__init__(acc_no, amount)
+        self.main_acc_instance = main_acc_instance
+```
+
+```
+saitama_savings_acc = SavingsAccount('ssa1234', 0, saitama_acc)
+```
+
+```
+saitama_acc.deposit(500)
+saitama_acc.withdraw(320)
+```
+
+```
+saitama_savings_acc.deposit(500)
+saitama_savings_acc.withdraw(320)
+```
+
+### super()
+In a child class, a parent class can be referenced by using the `super()` function. This super function returns a temporary object/instance of the superclass (parent class) that gives access to all its methods to its child class. There's no need to specify the parent class name to access its methods. This function can be used both in single and multiple inheritances.
```

Comment:
Always remember that we are not training parrots, we are training coders. How might a coder demonstrate this understanding? Can you come up with examples?